### PR TITLE
Setup styled components

### PR DESCRIPTION
**Pull request by codeAesthetic**
to be merged after pull request #3

### CODE REVIEW

**Comment by chirgjn** [comment]
Please request review only after dependent PRs are checked in

**Review by chirgjn — CHANGES_REQUESTED** [review]
> I think we can add the "Auth" related changes in a separate PR to keep this one more focused.

I added "Auth" just to showcase a basic example related to the styled-components.

**Comment by codeAesthetic** [comment]
I don't think we need to make a constant for magic number in CSS I mean usually I see it for javascript, not for CSS. For example, do I need to make constant just to store zero for global styles that don't make much sense.

**Comment by codeAesthetic** [comment]
> I think we can add the "Auth" related changes in a separate PR to keep this one more focused.

I added "Auth" just to showcase a basic example related to the styled-components.
sorry I edited your comment by mistake

**Comment by chirgjn** [comment]
https://github.com/pesto-students/juntos-frontend/pull/4#issuecomment-856956683

We don't want to pull-out numbers that don't have meaning attached to them, but you'd want to define some units / scale in your constants file for stuff like padding, margin etc.

A normal scale can be: 2px, 4px, 6px, 8px, 12px, 14px, 16px, 18px, 20px, 24px, 32px etc

**Comment by chirgjn** [comment]
https://github.com/pesto-students/juntos-frontend/pull/4#issuecomment-856959452

I think `Button` and `Input` are good as examples here. "Auth" is a bit too big.

File: src/components/Input.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import styled from "styled-components";
+
+export default styled.input`
+  background-color: #eee;
```
Comment (chirgjn) [inline src/components/Input.tsx:4]: We can extract out this colour

File: src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export default styled.button<ButtonProps>`
+  border-radius: 20px;
+  border: 1px solid ${colors.primaryOne};
+  background-color: ${colors.primaryOne};
+  color: ${colors.white};
+  font-size: 12px;
+  font-weight: bold;
+  padding: 12px 46px;
```
Comment (chirgjn) [inline src/components/Button.tsx:15]: Magic numbers

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import styled, { css } from "styled-components";
+import { colors } from "../common/colors";
+
+interface ButtonProps {
+  ghost?: boolean;
+}
+
+export default styled.button<ButtonProps>`
+  border-radius: 20px;
```
Comment (chirgjn) [inline src/components/Button.tsx:9]: Magic numbers

File: src/components/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  border: none;
+  padding: 12px 15px;
+  margin: 8px 0;
```
Comment (chirgjn) [inline src/components/Input.tsx:7]: Magic numbers

File: src/components/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  width: 100%;
+  border-radius: 4px;
```
Comment (chirgjn) [inline src/components/Input.tsx:9]: Magic numbers

**Review by chirgjn — CHANGES_REQUESTED** [review]

File: src/common/constants/cssScale.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  c3: "12px",
+  c4: "16px",
+  c5: "20px",
+  c6: "24px",
+  c7: "28px",
+  c8: "32px",
+  c9: "36px",
+  c10: "40px",
+  c11: "44px",
+};
```
Comment (chirgjn) [inline src/common/constants/cssScale.ts:14]: This works for now. For future projects you can reference: https://www.lightningdesignsystem.com/design-tokens/#category-spacing

**Review by chirgjn — APPROVED** [review]

**Comment by chirgjn** [comment]
Thanks for accommodating all the changes!

**Pull request merged**